include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -307,6 +309,43 @@ struct drop_view_simple
     }
 };
 
+template <typename _R, typename _Size>
+struct replicate_start_view_simple
+{
+    _R __r;
+    _Size __n;
+
+    replicate_start_view_simple(_R __rng, _Size __size) : __r(__rng), __n(__size)
+    {
+        assert(__n >= 0 && __n < __r.size());
+    }
+
+    //TODO: to be consistent with C++ standard, this Idx should be changed to diff_type of underlying range
+    template <typename Idx>
+    auto operator[](Idx __i) const -> decltype(__r[__i])
+    {
+        return (__i < __n) ? __r[0] : __r[__i - __n];
```

Comment:
What's exactly semantic of `replicate_start_view_simple` ?
Does it returns r[0] for any i? If yes, why do we process a case when `i >= n` ? m.b. better to wrute assert, like "i >= 0 and i < n"?

Reply:
I've fixed the assert to check that the number of replicates is not negative and that the incoming range is not empty.